LAPACKE/CMakeLists.txt

```diff
@@ -14,11 +19,6 @@ endif()
 add_subdirectory(include)
 
 
-if(NOT LAPACKE)
-    return()
-endif()
-
-
```

Comment:
> I am just unsure whether this specific displacement of code could cause problems using other configurations with cmake.

The code was specifically placed this way for #632.

Reply:
I agree with your suggestion, @friedc. I see no need to check the compatibility between Fortran and C compilers if we are just installing a C header file.

```diff
diff --git a/LAPACKE/CMakeLists.txt b/LAPACKE/CMakeLists.txt
index bbe84cbb2..ba105dfb2 100644
--- a/LAPACKE/CMakeLists.txt
+++ b/LAPACKE/CMakeLists.txt
@@ -1,12 +1,5 @@
-if(NOT LAPACKE)
-    return()
-endif()
-
-
 # Create a header file lapacke_mangling.h for the routines called in my C programs
 include(FortranCInterface)
-## Ensure that the fortran compiler and c compiler specified are compatible
-FortranCInterface_VERIFY()
 FortranCInterface_HEADER(${LAPACK_BINARY_DIR}/include/lapacke_mangling.h
   MACRO_NAMESPACE "LAPACK_"
   SYMBOL_NAMESPACE "LAPACK_")
@@ -18,6 +11,12 @@ endif()
 
 add_subdirectory(include)
 
+if(NOT LAPACKE)
+    return()
+endif()
+
+## Ensure that the fortran compiler and c compiler specified are compatible
+FortranCInterface_VERIFY()
 
 message(STATUS "LAPACKE enabled")
 enable_language(C)
 ```
 
This should be the patch, right? If so, I can propose a PR with this change to close #692.

Reply:
Acutally when trying

```
cmake -Bobjdir -DCMAKE_C_COMPILER=gcc -DCMAKE_Fortran_COMPILER=xlf -DBUILD_INDEX64=ON -DLAPACKE=ON
```

now it works without a problem

```
...
-- Detecting Fortran/C Interface
-- Detecting Fortran/C Interface - Found GLOBAL and MODULE mangling
-- Verifying Fortran/C Compiler Compatibility
-- Verifying Fortran/C Compiler Compatibility - Success
-- LAPACKE enabled
```

It would be easier to just revert 8d3ae005a3587e3e271cf80871d708d6b38c7fff since no one else is having a problem with it.